schemas/typedspans/README.md

```diff
@@ -0,0 +1,278 @@
+
+# Typed Spans
+
+## Rationale
+
+Spans represent different _canonical_ _types _of operations. Examples are
+
+* Local operations like method invocations
+* HTTP requests (inbound and outbound)
+* Database requests
+* eneric RPC requests like GRPC
+
+A span consists of a number of mandatory and optional attributes that add information
+about the represented operation to it.
+
+Depending on the _canonical type_ of an operation some attributes might be needed
+to represent and analyze a span correctly.
+
+**Example:** A HTTP request needs a status code to distinguish successful or
+unsuccessful operations.
+
+Right now, spans can be created freely and it’s up to the implementor to set all
+the attributes needed to represent the given type.
+
+## Existing work
+
+* There is already a _Type_ field that describes if a span is a parent or child span
+* There is already a _SpanKind_ field that describes if a span is a server or
+  client span - this distinguishes inbound and outbound requests.
+* There should be already a TypedSpanBuilder for Java
+
+## Proposal
+
+* Implement a **TypedSpanBuilder** for all platforms that ensures that all
+  needed attributes and the type is set.
+* Add a field **CanonicalType** that contains the type of span.
```

Comment:
```suggestion
* Add a field `CanonicalType` that contains the type of span.
```

Reply:
I'd suggest to add more details on what canonical means. For instance, for the database call - will type be `db`, `sql`, `postgreSQL`?